OpenRA.Game/Game.cs

```diff
@@ -955,15 +955,18 @@ public static void LoadMap(string launchMap)
 				Order.Command("option gamespeed default"),
 				Order.Command($"state {Session.ClientState.Ready}")
 			};
+			try
+			{
+				var package = new OpenRA.FileSystem.Folder(Platform.EngineDir).OpenPackage(launchMap, ModData.ModFiles);
+				var map = ModData.MapCache.SingleOrDefault(m => m.Uid == launchMap || m.Uid == Map.ComputeUID(package));
 
-			var path = Platform.ResolvePath(launchMap);
-			var map = ModData.MapCache.SingleOrDefault(m => m.Uid == launchMap) ??
-				ModData.MapCache.SingleOrDefault(m => m.Package.Name == path);
-
-			if (map == null)
-				throw new InvalidOperationException($"Could not find map '{launchMap}'.");
-
-			CreateAndStartLocalServer(map.Uid, orders);
+				CreateAndStartLocalServer(map.Uid, orders);
+			}
+			catch (System.NullReferenceException)
+			{
+				Console.WriteLine($"Invalid Launch.Map parameter given. No map found at {launchMap}. Exiting program.");
+				Exit();
+			}
```

Comment:
This `try catch log` pattern is only very seldom used in OpenRA and not for trivial stuff like a null reference exception that should certainly be handled. This seems like a needless change, but if you insist, I suggest:

```csharp
throw new ArgumentException($"Could not find map '{launchMap}'.");
```

If the game does not start, you certainly shouldn't `Exit()` without an exit code that indicates the error.

Reply:
I don't think we should use try-catch here. What is the problem with checking `package` for `null` before computing the id?

Reply:
Hi @Mailaender , @abcdefg30 , thanks for the feedback, have replaced the try/catch with a 'throw new ArgumentException', by checking whether the package variable is null before attempting to use ComputeUID(), as was recommended by abcdefg30. I can confirm that with this change ComputeUID no longer results in a NullPointerException, while still exiting out via an exception if an invalid map param was provided. Have also tested again loading maps with a UID, absolute path, and relative path, and all are working as before. Please let me know if you need any further changes to the code.